system/Config/Factories.php

```diff
@@ -125,18 +205,27 @@ public static function __callStatic(string $component, array $arguments)
     protected static function locateClass(array $options, string $name): ?string
     {
         // Check for low-hanging fruit
-        if (class_exists($name, false) && self::verifyPreferApp($options, $name) && self::verifyInstanceOf($options, $name)) {
+        if (
+            class_exists($name, false)
+            && self::verifyPreferApp($options, $name)
+            && self::verifyInstanceOf($options, $name)
+        ) {
             return $name;
         }
 
         // Determine the relative class names we need
         $basename = self::getBasename($name);
-        $appname  = $options['component'] === 'config'
+        $appname  = self::isConfig($options['component'])
             ? 'Config\\' . $basename
             : rtrim(APP_NAMESPACE, '\\') . '\\' . $options['path'] . '\\' . $basename;
 
         // If an App version was requested then see if it verifies
-        if ($options['preferApp'] && class_exists($appname) && self::verifyInstanceOf($options, $name)) {
+        if (
+            // preferApp is used only for no namespace class.
+            strpos($name, '\\') === false
```

Comment:
Is this the only breaking change? It seems like the rest of this PR could be rolled out separately without as big an impact.

Reply:
Perhaps. 

And it is what I want to change. Also it is Paul's opinion. See https://github.com/codeigniter4/CodeIgniter4/issues/7694#issuecomment-1631110862